azure-pipelines.yml

```diff
@@ -145,7 +145,7 @@ stages:
           steps:
           - checkout: self
             clean: true
-          - script: eng/cibuild.sh --configuration $(_BuildConfig) --prepareMachine
+          - script: eng/cibuild.sh --configuration $(_BuildConfig) --prepareMachine /p:SourceBuildIntermediateNupkgRid='linux-x64'
```

Comment:
Added since `runtime`'s intermediate is not `ManagedOnly`

Reply:
What does this property do and why isn't it set in `common/cibuild.sh`?

Reply:
The property specifies to source-build MSBuild logic the RID of an intermediate package. In cases where intermediate packages are platform-specific (like with `runtime`), this property has to be set or the source-build will fail to `restore`. In this case, we are restoring a portable intermediate, hence the `linux-x64` value.

My bad, I did not notice there was a different script in `eng`, will put it there instead.

As for `common/cibuild.sh` - it is coming from Arcade, so any change there would get overridden / conflict once a new Arcade version flows in, no?

Reply:
Why is this property needed for this repo and not other repos?

Reply:
No other repository that I have worked with was using a runtime intermediate. We are on our way to enable pre-built detection in repos, so most probably others that are dependent on latest runtime would get a similar property

Reply:
Pardon my ignorance, but what does that mean that the "runtime intermediate" is being used? How does that happen?

Reply:
Intermediate packages are a transport format we use to transfer source-built pre-builts between repositories ([docs](https://github.com/dotnet/source-build/blob/main/Documentation/planning/arcade-powered-source-build/intermediate-nupkg.md)).

In this case, during SourceLink's source-build the build logic will retrieve the runtime's source-build by restoring the intermediate packages and use assets from there in the build. These assets can be platform-specific, hence there are different packages for different platforms. For the build logic to know which exact intermediate to pull, we need to specify the `SourceBuildIntermediateNupkgRid` property

Reply:
You're saying:
> No other repository that I have worked with was using a runtime intermediate. 

I'm still not understanding what's special about Source Link build that would be different from any other repository that builds a dotnet library.

Reply:
Answered here: https://github.com/dotnet/sourcelink/pull/933#discussion_r1068242626

Reply:
Removed with the new changes from Arcade (SB build logic will identify the RID param by itself)